src/platform/PICO/bus_quadspi_pico.c

```diff
@@ -0,0 +1,371 @@
+        .bk1IO3Pins = { { 0 } },
+        .bk1CSPins  = { { 0 } },
+        .bk2IO0Pins = { { 0 } },
+        .bk2IO1Pins = { { 0 } },
+        .bk2IO2Pins = { { 0 } },
+        .bk2IO3Pins = { { 0 } },
+        .bk2CSPins  = { { 0 } },
+    },
+};
+
+// Pin configure is a no-op on RP2350, QMI pins are fixed and controlled by ROM/QMI
+void quadSpiPinConfigure(const quadSpiConfig_t *pConfig)
+{
+    UNUSED(pConfig);
+    quadSpiDevice[QUADSPIDEV_1].dev = quadSpiHardware[0].reg;
+}
+
+static inline void pico_qspi_enter_cmd_mode(void)
+{
+    // Ensure XIP cache is clean before exiting XIP
+    xip_cache_clean_all();
+    rom_connect_internal_flash_fn connect_fn = (rom_connect_internal_flash_fn)rom_func_lookup_inline(ROM_FUNC_CONNECT_INTERNAL_FLASH);
+    rom_flash_exit_xip_fn exit_fn = (rom_flash_exit_xip_fn)rom_func_lookup_inline(ROM_FUNC_FLASH_EXIT_XIP);
+    connect_fn();
+    exit_fn();
+}
+
+static inline void pico_qspi_exit_cmd_mode(void)
+{
+    flash_flush_cache();
+}
+
+// QMI direct helpers for CS1 transactions
+static inline void qmi_direct_enable(void) { hw_set_bits(&qmi_hw->direct_csr, QMI_DIRECT_CSR_EN_BITS); }
+static inline void qmi_direct_disable(void) { hw_clear_bits(&qmi_hw->direct_csr, QMI_DIRECT_CSR_EN_BITS); }
+static inline void qmi_cs1_assert(bool asserted) {
+    if (asserted) hw_set_bits(&qmi_hw->direct_csr, QMI_DIRECT_CSR_ASSERT_CS1N_BITS);
+    else hw_clear_bits(&qmi_hw->direct_csr, QMI_DIRECT_CSR_ASSERT_CS1N_BITS);
+}
+
+static inline void qmi_timeout_cleanup(void)
+{
+    qmi_cs1_assert(false);
+    qmi_direct_disable();
+    pico_qspi_exit_cmd_mode();
+}
+
+// Default timeout for QSPI direct IO operations
+#define QSPI_TIMEOUT_MS 10
+
+// Number of dummy clock cycles per consumed byte in direct-mode loops.
+// Justification: QSPI datasheets specify dummy cycles in bits. Our direct-mode
+// helper consumes dummy time by shifting out whole bytes (8 bits at a time).
+// We therefore convert from bits to bytes using 8 bits/byte. If a controller
+// revision were to support sub-byte dummy handling, this could be revisited.
+#define QSPI_DUMMY_BITS_PER_BYTE 8
+
+static inline bool qmi_direct_io(const uint8_t *tx, uint8_t *rx, size_t count, timeMs_t timeout)
+{
+    size_t tx_remaining = count;
+    size_t rx_remaining = count;
+    // Prevent hangs: break out after timeout in ms
+    const timeMs_t start_ms = millis();
+    while (tx_remaining || rx_remaining) {
+        uint32_t flags = qmi_hw->direct_csr;
+        bool can_put = !(flags & QMI_DIRECT_CSR_TXFULL_BITS);
+        bool can_get = !(flags & QMI_DIRECT_CSR_RXEMPTY_BITS);
+
+        if (can_put && tx_remaining) {
+            qmi_hw->direct_tx = tx ? *tx++ : 0x00;
+            --tx_remaining;
+        }
+        if (can_get && rx_remaining) {
+            uint8_t b = (uint8_t)qmi_hw->direct_rx;
+            if (rx) {
+                *rx++ = b;
+            }
+            --rx_remaining;
+        }
+        if (cmpTimeMs(millis(), start_ms) > (int32_t)timeout) {
```

Comment:
(Maybe pass timeout as `int` / `int32_t` ?)

Reply:
Yeah, it's better to pass `timeout` as `uint32_t` rather than `timeMs_t` which represents an absolute number of ms.